util/csv_util.ts

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env ts-node-script
+
+import { parseString } from '@fast-csv/parse';
+import fetch from 'node-fetch';
+import { Lang } from '../resources/languages';
+
+const baseGithub = 'https://raw.githubusercontent.com/';
+const intlGithub = 'xivapi/ffxiv-datamining/master/csv/';
+const cnGithub = 'thewakingsands/ffxiv-datamining-cn/master/';
+const koGithub = 'Ra-Workspace/ffxiv-datamining-ko/master/csv/';
+
+
+export type Table<RowKey extends string, SubKey extends string | number> =
+  Record<RowKey extends '#' ? number : string, Record<RowKey | SubKey, string | undefined>>;
+
+export type GetTableFunc = {
+  <T extends (string | number), K extends string>(
+    table: string, inputs: [key: K, ...indices: T[]],
+  ): Promise<Table<K, T>>;
+  <T extends (string | number), K extends string>(
+    table: string, inputs: (string | number)[], outputs?: [key: K, ...indices: T[]],
+  ): Promise<Table<K, T>>;
+};
+
+// Turn names from tables into JavaScript-safe ascii string keys.
+export const cleanName = (str?: string | null): string | undefined | null => {
```

Comment:
What calls this with null?

Reply:
There might be some null value in CSVs I think?

Reply:
I think everything is a string (possibly the empty string, but still a string).

Reply:
Sure, but then you should always make sure the parameter is not null before passed into the function.